Add new terms to the glossary (#20051)
huggingface/transformers: docs/source/en/glossary.mdx

embeddings that are computed in parallel and thus defines the trade-off between memory and time complexity. If
`chunk_size` is set to 0, no feed forward chunking is done.

## H

### head

The model head refers to the last layer of a neural network that accepts the raw hidden states and projects them onto a different dimension. There is a different model head for each task. For example:

  * [`GPT2ForSequenceClassification`] is a sequence classification head which is a linear layer on top of the base [`GPT2Model`].

>   * [`GPT2LMHeadModel`] is a language modeling head which is a linear layer with weights tied to the input embeddings on top of the base [`GPT2Model`].

Reviewer: This one is not a good example actually, as `LMHead` is very ambiguous. Can you use a `ForCausalLM` or `ForMaskedLM` instead?

Also maybe give examples in speech and vision (like `ForObjectClassification` or `ForCTC`.

predictions and the expected value (the label).

These labels are different according to the model head, for example:

- For sequence classification models ([`BertForSequenceClassification`]), the model expects a tensor of dimension
  `(batch_size)` with each value of the batch corresponding to the expected label of the entire sequence.
- For token classification models ([`BertForTokenClassification`]), the model expects a tensor of dimension
  `(batch_size, seq_length)` with each value corresponding to the expected label of each individual token.
- For masked language modeling ([`BertForMaskedLM`]), the model expects a tensor of dimension `(batch_size,
  seq_length)` with each value corresponding to the expected label of each individual token: the labels being the token
  ID for the masked token, and values to be ignored for the rest (usually -100).
- For sequence to sequence tasks,([`BartForConditionalGeneration`], [`MBartForConditionalGeneration`]), the model
  expects a tensor of dimension `(batch_size, tgt_seq_length)` with each value corresponding to the target sequences
  associated with each input sequence. During training, both BART and T5 will make the appropriate

>   `decoder_input_ids` and decoder attention masks internally. They usually do not need to be supplied. This does not

Reviewer: Would be great to expand this section for other modalities as well (can be a followup PR!). And also leave one last remark telling the user to always check the model documentation to be sure what the label are called.

Author: Great suggestion, I'll add it to my list!

## N

### Natural language generation

All tasks related to generating text (for instance talk with transformers, translation).

### Natural language processing

A generic way to say "deal with texts".

### Natural language understanding

All tasks related to understanding what is in a text (for instance classifying the
whole text, individual words).

## P

### pixel values

A tensor of the numerical representations of an image that is passed to a model. The pixel values have a shape of [`batch_size`, `num_channels`, `height`, `width`], and are generated from a feature extractor.

### pooling

> An operation that reduces a matrix into a smaller matrix, either by taking the maximum or average of the pooled area. Pooling is similar to [convolution](#convolution) in the sense that it is repeated over the entire matrix, and each operation is applied to a different segment of the matrix.

Reviewer: The second sentence is very confusing. Pooling is very different from convolution, which is applied on windows of the matrix, whereas pooling is applied on one (or several) dims. 
```suggestion
An operation that reduces a matrix into a smaller matrix, either by taking the maximum or average of the pooled dimension(s).
```

### pretrained model

A model that has been pretrained on some data (for instance all of Wikipedia). Pretraining methods involve a
self-supervised objective, which can be reading the text and trying to predict the next word (see [causal language
modeling](#causal-language-modeling)) or masking some words and trying to predict them (see [masked language

> modeling](#masked-language-modeling)).

Reviewer: This is too text-centric as well. Other modalities have different pretraining objectives and it would be great to mention them here.
